src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -481,22 +493,99 @@ private string GetCloudApplicationName(string? applicationName, UserDeploymentSe
+
+            var userResponse =
+                        _consoleUtilities.AskUserToChooseOrCreateNew(
+                            deployedApplications.Select(x => x.DisplayName),
+                            title,
+                            askNewName: false,
+                            defaultChoosePrompt: Constants.CLI.PROMPT_CHOOSE_DEPLOYMENT_TARGET,
+                            defaultCreateNewLabel: Constants.CLI.CREATE_NEW_APPLICATION_LABEL);
+
+            if (userResponse.CreateNew)
+                return string.Empty;
+
+            // Since the selected option will be the display name, we need to extract the actual name out of it.
+            // Ex - DisplayName = "MyAppStack (CloudFormationStack)", ActualName = "MyAppStack"
+            var cloudApplicationName = userResponse.SelectedOption?.Split().FirstOrDefault() ?? string.Empty;
+            return cloudApplicationName;
+        }
+
+
+        // This method prompts the user for a new CloudApplication name and also generate a valid default name by respecting existing applications.
+        private string AskForNewCloudApplicationName(DeploymentTypes deploymentType, List<CloudApplication> deployedApplications)
+        {
             if (_toolInteractiveService.DisableInteractive)
             {
                 var message = "The \"--silent\" CLI argument can only be used if a cloud application name is provided either via the CLI argument \"--application-name\" or through a deployment-settings file. " +
                 "Please provide an application name and try again";
                 throw new InvalidCliArgumentException(DeployToolErrorCode.SilentArgumentNeedsApplicationNameArgument, message);
             }
-            return AskUserForCloudApplicationName(_session.ProjectDefinition, deployedApplications);
+
+            var defaultName = "";
+
+            try
+            {
+                defaultName = _cloudApplicationNameGenerator.GenerateValidName(_session.ProjectDefinition, deployedApplications);
+            }
+            catch (Exception exception)
+            {
+                _toolInteractiveService.WriteDebugLine(exception.PrettyPrint());
+            }
+
+            var cloudApplicationName = "";
+
+            while (true)
+            {
+                _toolInteractiveService.WriteLine();
+
+                var title = "Name the Cloud Application to deploy your project to" + Environment.NewLine +
+                            "--------------------------------------------------------------------------------";
+
+                string inputPrompt;
+
+                switch (deploymentType)
+                {
+                    case DeploymentTypes.CdkProject:
+                        inputPrompt = Constants.CLI.PROMPT_NEW_STACK_NAME;
+                        break;
+                    default:
+                        throw new InvalidOperationException($"The {nameof(DeploymentTypes)} {deploymentType} does not have an input prompt");

```

Comment:
I'd refrain from using non-deploy tool exceptions which will lead to an ugly stack trace returned to the user if thrown.

Reply:
The deploy-tool exceptions are presented only in case of a user initiated error. They are used to display errors which can be fixed by the user directly. In this case, the error will not occur due to a user action. Its an implementation bug and we want to fail with a full stack trace.

Reply:
Sounds good